### DISCUSSION TITLE
Combining priceanalyzer with home/away sensor and grid conspumtion limitation

### DISCUSSION THREAD

**Discussion by TurboJens** [erlendsellie/priceanalyzer#32] [Ideas]
Hello :)

I find this tool very helpful, but for my (and maybe many others) I would like a low entry possibility to combine priceanalyzer with home/away sensor as I do travel a lot in work, and whem I'm not home I really don't need any hot water at all, and when returning my heater is so fast that it will be heated up quite fast. I plan tweaking this by setting the zone radius to aprox 2h from home.

It would also be great to combine this with a sensor for stopping the hot water heater to keep usage within certain energy grid tariffs.

The way I thougt this could be implemented is by having two extra options in the automation blueprint, one for disabling when away and one for disabling when consumtion within the current hours is predicted to pass the next tariff level. And that this is optional, if nothing is spesified nothing will happen.
![image](https://github.com/erlendsellie/priceanalyzer/assets/76071330/a2f34eae-e183-428a-9bb3-7fffb6853822)

Best regards,
Jens

**Comment by erlendsellie** [comment]
Thanks for the feature request!
This would indeed be nice, and is on the todo-list, but maybe as a service-call instead of through the blueprint.
I am not using the blueprint myself, but rather pass the VVBsensor from PriceAnalyzer through a new Template Sensor in HA, which controls my heater.
This template sensor also listens for some other input, like 'input_boolean.limit_water_heater' in the case that my grid consumption limitation decide's it's time to limit the water heater.
I've also got a few input_booleans in the UI for the exact reason you mention. I can toggle one to keep it on, or one to keep it off. I also got one toggle to keep it off - if this one is not on, it will turn back on the next day.

I don't have the spare time to develop this in the integration at the moment, but if you want to implement anything like this, my VVB-code can be found here (with a link to the template sensor, for example: [https://github.com/erlendsellie/HomeAssistantConfig/blob/12f31120324d56de27d6f7e74d5464da7c793ed6/packages/power/vvb.yaml#L54](https://github.com/erlendsellie/HomeAssistantConfig/blob/12f31120324d56de27d6f7e74d5464da7c793ed6/packages/power/vvb.yaml#L54))

**Reply by TurboJens** [reply]
Thanks for your reply.

At this moment I have a almost fully working flow in node-red for this, that based on priceanalyzers setpoint control water heater and it shuts off if power consumption is too high (goal is to consume less than 2kWh and water heater is 2.8kW) it also reduce the temperature when nobody is home and nobody is closer to home than about 1.5hour.

Seems to work very well at this moment, need some fine tuning only. Node-red is quite confusing though so it's not the easiest task to accomplish :)

**Reply by erlendsellie** [reply]
Sounds like you went the same route that i went then.

I did almost the same thing with the setpoint-temperature in YAML, but based on kilometers away.
Here is my effect-control automation/yaml, if it may help. I can change an input_number, and stay under 2kWh/h if i want, but i'll end up with an empty battery on the car, empty water heater and a cold house, hehe.

https://github.com/erlendsellie/HomeAssistantConfig/blob/master/packages/effectcontrol.yaml

**Reply by TurboJens** [reply]
I have no electric vehicles and all heating is underfloor water heating where energy is delivered from condominium. And even the water heater is preheated to 30-35 degrees, so there is no big issue to keep consumption less than 2kWh, the only thing is that numbers is very low, so I have to put a bit of effort in it so I don't run all appliances at the same time as washing machine, dishwasher, hob and such things easily breaks it. Having control of water heater helps a lot, since it's else is living it's own life and makes it impossible.

I could easily just keep within 2-5kW and keep it simple, but thats not fun :)
